### PR TITLE
fix: remove extraneous mime-types package in favor of mime

### LINKED ISSUE #2152: refactor: remove mime-types / mime-db dependency
Currently we utilize `mime` to detect content types and set metadata accordingly. Mime uses mime-db which is essentially a lookup table from JSON files. This dependency is accounting for ~13% of our package size. We should remove this dependency in favor of something smaller for example we could write a MIME sniffer based on the [spec](https://mimesniff.spec.whatwg.org/#matching-a-mime-type-pattern) if one does not already exist for Node.

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #2152 🦕

### CODE REVIEW

**Review by danielbankhead — APPROVED** [review]

**Comment by ddelgrosso1** [comment]
Going to add the additional logic to remove `compressible`. Adding do not merge in the meantime.

**Comment by ddelgrosso1** [comment]
Added additional logic and removed compressible from the dependencies.

File: src/file.ts [hunk 1]
```diff
@@ -326,6 +325,24 @@ export const STORAGE_POST_POLICY_BASE_URL = 'https://storage.googleapis.com';
  */
 const GS_URL_REGEXP = /^gs:\/\/([a-z0-9_.-]+)\/(.+)$/;
 
+/**
+ * @private
```
Comment (danielbankhead) [inline src/file.ts:329]: Perhaps a comment or [@privateRemarks](https://tsdoc.org/pages/tags/privateremarks/) on how to regenerate this regex over time would keep this maintainable in the future

**Review by danielbankhead — APPROVED** [review]
LGTM, one comment

**Pull request merged**